azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java

```diff
@@ -482,13 +482,56 @@
     }
 
     /**
+     * The virtual machine definition stage with availability set.
+     *
+     * @param <T> The virtual machine definition in creatable stage.
+     */
+    interface DefinitionWithAvailabilitySet<T extends DefinitionCreatable> {
+        /**
+         * Specifies the name of the availability set to create, the virtual machine will be part of
+         * this availability set.
+         * <p/>
+         * Adding virtual machines running your application to an availability set ensures that during
+         * maintenance event at least one virtual machine will be available.
+         *
+         * @param name The name of the availability set
+         * @return The stage representing creatable VM definition
+         */
+        T withNewAvailabilitySet(String name);
+
+        /**
+         * Specifies an instance of AvailabilitySet.DefinitionCreatable representing the availability set
+         * to be created, the virtual machine will be part of this availability set.
```

Comment:
When referencing a type or a type member, it'd be good to do this: {@link AvailabilitySet.DefinitionCreatable}, which creates a link to that Javadoc.  
As for the wording, I'm wondering maybe we can be more descriptive of what's going on, rather than merely saying the technical detail. Maybe like: "Specifies the definition of an not-yet-created availability set to associate the virtual machine with.<p>
The definition needs to be at the {@link AvailabilitySet.DefinitionCreatable} stage."